ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java

```diff
@@ -60,13 +61,12 @@ public class SimpleStateMachineStorage implements StateMachineStorage {
       Pattern.compile(SNAPSHOT_FILE_PREFIX + "\\.(\\d+)_(\\d+)");
 
   private volatile File stateMachineDir = null;
-
-  private volatile SingleFileSnapshotInfo currentSnapshot = null;
+  private final AtomicReference<SnapshotInfo> latestSnapshot = new AtomicReference<>();
```

Comment:
If we use `AtomicReference<SingleFileSnapshotInfo>` here, we don't have to cast in the caller site if we're sure that the snapshot storage is `SimpleStateMachineStorage`.

Reply:
This is a good idea.  Thanks!